docs/design/autoware-interfaces/ad-api/types/autoware_ad_api_msgs/msg/lanelet_primitive.md

```diff
@@ -0,0 +1,12 @@
+# autoware_ad_api_msgs/msg/LaneletPrimitive
+
+## Definition
+
+```txt
+int64 id
+string type
```

Comment:
Do we need this type? I guess it's in the map binary and it seems not being used.
@mitsudome-r @xmfcx What do you think?

Reply:
FYI, it was used to switch between lane planning mode and free space planning mode in Autoware.Auto.

If we can deduce this by accessing the map, I think we can remove it.
However, current lanelet2 does not forbid setting same id if primitive types are different (e.g., you can have `id=1` area and `id=1` lanelet) so if we would like to have route to include both area types and lanelet we either have to:
* add id in route message
* make additional specification as Autoware that we only support maps with unique ids.

Reply:
I see, thank you for your information.
If `id` is not unique, then it's okay for me to keep `type`.

Reply:
So will we add a comment like this?

```suggestion
string type # We need to specify the type in order to uniquely identify a primitive. Lanelet2 does not forbid setting the same id if primitive types are different.
```

Reply:
Added a comment simply. https://github.com/autowarefoundation/autoware-documentation/commit/211737c245b8ca0f49ab66f84b7693f74afc831d.
> The same id may be used for each type.